src/picongpu/include/plugins/particleDetector/ParticleDetector.hpp

```diff
@@ -0,0 +1,542 @@
+                      << std::endl;
+            return;
+        }
+
+        this->maxTheta = float_X(0.5) * this->openingTheta;
+        this->maxPhi = float_X(0.5) * this->openingPhi;
+        /* convert units */
+        const float_64 minEnergy_SI = this->minEnergy * UNITCONV_keV_to_Joule;
+        const float_64 maxEnergy_SI = this->maxEnergy * UNITCONV_keV_to_Joule;
+        this->minEnergy = minEnergy_SI / UNIT_ENERGY;
+        this->maxEnergy = maxEnergy_SI / UNIT_ENERGY;
+
+        /* allocate memory buffers */
+        this->dBufCalorimeter = new DBufCalorimeter(this->numBinsTheta, this->numBinsPhi, this->numBinsEnergy);
+        this->dBufLeftParsCalorimeter = new DBufCalorimeter(this->dBufCalorimeter->size());
+        this->hBufCalorimeter = new HBufCalorimeter(this->dBufCalorimeter->size());
+        this->hBufTotalCalorimeter = new HBufCalorimeter(this->dBufCalorimeter->size());
+
+        /* fill calorimeter for left particles with zero */
+        this->dBufLeftParsCalorimeter->assign(float_X(0.0));
+
+        /* create mpi reduce algorithm */
+        PMacc::GridController<simDim>& con = PMacc::Environment<simDim>::get().GridController();
+        pm::Size_t<simDim> gpuDim = (pm::Size_t<simDim>)con.getGpuNodes();
+        zone::SphericZone<simDim> zone_allGPUs(gpuDim);
+        this->allGPU_reduce = AllGPU_reduce(new PMacc::algorithm::mpi::Reduce<simDim>(zone_allGPUs));
+
+        /* calculate rotated detector frame from posTheta and posPhi */
+        const float_64 posTheta_rad = this->posTheta * float_64(M_PI / 180.0);
+        const float_64 posPhi_rad = this->posPhi * float_64(M_PI / 180.0);
+        this->detectorFrameVecY = float3_X(pam::sin(posTheta_rad) * pam::cos(posPhi_rad),
+                                           pam::cos(posTheta_rad) * pam::cos(posPhi_rad),
+                                           pam::sin(posPhi_rad));
+        /* If the y-axis is pointing exactly up- or downwards we need to define the x-axis manually */
+        if(pam::abs(this->detectorFrameVecY) == float_X(1.0))
+        {
+            this->detectorFrameVecX = float3_X(1.0, 0.0, 0.0);
+        }
+        else
+        {
+            const float3_X vecUp(0.0, 0.0, -1.0);
+            this->detectorFrameVecX = pam::cross(vecUp, this->detectorFrameVecY);
+            /* normalize vector */
+            this->detectorFrameVecX /= pam::abs(this->detectorFrameVecX);
+        }
+        this->detectorFrameVecZ = pam::cross(this->detectorFrameVecX, this->detectorFrameVecY);
+
+        /* create calorimeter functor instance */
+        this->calorimeterFunctor = MyCalorimeterFunctorPtr(new MyCalorimeterFunctor(
+            this->maxTheta * float_X(M_PI / 180.0),
+            this->maxPhi * float_X(M_PI / 180.0),
+            this->numBinsTheta,
+            this->numBinsPhi,
+            this->numBinsEnergy,
+            this->logScale ? pam::log10(this->minEnergy) : this->minEnergy,
```

Comment:
Please add a comment, that in case  of a logarithmic scale your `minEnergy` and `maxEnergy` is already converted to log values.